packages/pug-code-gen/index.js

```diff
@@ -767,6 +768,37 @@ Compiler.prototype = {
     this.buf.push('  }\n}).call(this);\n');
   },
 
+  visitEachOf: function(each){
+    var valVarName = 'pug_val' + this.eachOfCount;
+    if (each.key) {
+      var keyVarName = 'pug_key' + this.eachOfCount
+    }
+    this.eachOfCount++;
+
+    this.buf.push(''
+      + '// iterate ' + each.obj + '\n'
+      + ';(function(){\n'
+      + '  var $$obj = ' + each.obj + ';\n'
```

Comment:
We should be able to inline `each.obj` in the js `for` statement. It only gets evaluated once anyway, so there's no need to assign it to a variable.